### PR TITLE
mona requires respect when ordering water

### LINKED ISSUE #76: Mona requires respect
Mona is a bot but it doesn't mean that we can skip respectful communication

**Proposal**
When asking Mona to order water, `please` should be required

Example:
`@mona order water please`

### PR DESCRIPTION
**Pull request by lazycoder9**
Close #76 
Now you should add `please`, `pls` or `plz` if you want to order water. If not, your request will be declined and you will get something like this

image

### CODE REVIEW

File: specs/waterOrderingSpec.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ describe('Water ordering controller', () => {
     waterOrderingController(this.controller)
   })
 
-  it('returns any confirmation if user types `mona order water`', async () => {
+  it('declined order if no respect', async () => {
```
Comment (nikitasakau) [inline specs/waterOrderingSpec.js]: Maybe `declines`?

File: features/waterOrdering/index.js [hunk 2]
```diff
@@ -65,23 +59,37 @@ const passedEnoughTimeFromLastOrder = async (robot) => {
   return lastWaterOrder === null || isEnoughTimePassed(lastWaterOrder)
 }
 
+const withRespect = ({ text }) => text.toLowerCase().includes('please')
+
+const getResponsesList = async (bot, message) => {
+  const { good, sendingError, noRespect } = replies
```
Comment (nikitasakau) [inline features/waterOrdering/index.js]: If you decide to refactor this feature maybe it is good time to give normal names for these replies. `good` and `sendingError` sound like something strange. 🙂Thoughts?

File: specs/waterOrderingSpec.js (hunk 1 above)
Comment (nikitasakau) [reply specs/waterOrderingSpec.js]: Or even something with returns (kinda `returns any phrases asking for respect if no respect`) in order to have the same styles for tests

File: features/waterOrdering/index.js [hunk 3]
```diff
@@ -65,23 +59,37 @@ const passedEnoughTimeFromLastOrder = async (robot) => {
   return lastWaterOrder === null || isEnoughTimePassed(lastWaterOrder)
 }
 
+const withRespect = ({ text }) => text.toLowerCase().includes('please')
```
Comment (AleksSenkou) [inline features/waterOrdering/index.js]: Thoughts about adding `plz` and `pls` as well?

File: features/waterOrdering/index.js [hunk 4]
```diff
@@ -65,23 +59,37 @@ const passedEnoughTimeFromLastOrder = async (robot) => {
...
+  if (!withRespect(message)) {
+    return noRespect
+  }
+
+  if (passedEnoughTime) {
+    try {
+      await sendOrderToWaterDealer(bot)
+
+      return good
+    } catch (e) {
```
Comment (AleksSenkou) [inline features/waterOrdering/index.js]: what is `e`?

File: features/waterOrdering/replies.js [hunk 5]
```diff
@@ -10,6 +10,14 @@ module.exports = {
     'Done :white_check_mark:.',
     'Yes!',
   ],
+  noRespect: [
+    'Maybe "Please"? :grumpy:',
+    'You ask me for a service, but do it without respect? :godfather:',
```
Comment (AleksSenkou) [inline features/waterOrdering/replies.js:15]: Not sure that we need a question mark at the end of this message, thoughts?

File: features/waterOrdering/replies.js [hunk 6]
```diff
@@ -10,6 +10,14 @@ module.exports = {
...
+    'Actually, I deserve respect :dude:',
+    'How dare you not respect me? :accepted:',
+    'Water shall not pass!!! :not_pass:',
+    'Demand respect I :yoda:',
```
Comment (AleksSenkou) [inline features/waterOrdering/replies.js:19]: Can we somehow say that `please` is required in messages above?

Just not sure how to do better but would like people to understand that Mona is waiting for `please`

File: features/waterOrdering/replies.js (hunk 5 above)
Comment (lazycoder9) [reply features/waterOrdering/replies.js:15]: @AleksSenkou why do you think so? Mona is asking question

Comment (AleksSenkou) [reply features/waterOrdering/replies.js:15]: `You ask me for a service, but do you do it with a respect?` is a question

File: features/waterOrdering/index.js [hunk 7]
```diff
@@ -7,12 +7,21 @@ const { randomArrayItem } = require('../../lib')
...
   ))
 )
 
+const addRespectNotes = () => {
+  const { askRespect, respectNoteText } = replies
+
+  // I'm not sure how to properly concatenate note to reply to appear it in newline in Slack
+  // \n in Slack may not work
+  return askRespect.map(reply => `${reply}
+${respectNoteText}`)
```
Comment (AleksSenkou) [inline features/waterOrdering/index.js]: Is `\n` works here?

File: features/waterOrdering/replies.js [hunk 8]
```diff
@@ -10,7 +10,16 @@ module.exports = {
...
-  sendingError: [
+  askRespect: [
+    'Maybe "Please"? :grumpy:',
+    'You ask me for a service, but do it without respect? :godfather:',
+    'Actually, I deserve respect :dude:',
+    'How dare you not respect me? :accepted:',
+    'Water shall not pass!!! :not_pass:',
+    'Demand respect I :yoda:',
+  ],
+  respectNoteText: '_(you can show you respect with *please*, *pls* or *plz* at the end of message)_',
```
Comment (AleksSenkou) [inline features/waterOrdering/replies.js]: Thoughts about making it simpler?

`respectful requests include *please*, *pls* or *plz*`

File: features/waterOrdering/index.js [hunk 9]
```diff
@@ -7,12 +7,20 @@ const { randomArrayItem } = require('../../lib')
...
   replies.tooMuchOrdersError.map(reply => (
     reply.replace('{{lastWaterOrderCreatedAt}}', lastOrderTime)
   ))
 )
 
+const addRespectNotes = () => {
+  const { askRespect, respectNoteText } = replies
+
+  // I'm not sure how to properly concatenate note to reply to appear it in newline in Slack
+  // \n in Slack may not work
```
Comment (AleksSenkou) [inline features/waterOrdering/index.js]: So have you tested it in slack?

Comment (lazycoder9) [reply features/waterOrdering/index.js]: @AleksSenkou yeap, tested, here is a screenshots (sent messages from JS code)
image

Comment (AleksSenkou) [reply features/waterOrdering/index.js]: So why have you left these comments then? 😄 

```
// I'm not sure how to properly concatenate note to reply to appear it in newline in Slack
  // \n in Slack may not work
```

Comment (lazycoder9) [reply features/waterOrdering/index.js]: @AleksSenkou already done)

**Review by AleksSenkou — APPROVED** [review]
LGTM!

**Pull request merged**